### ISSUE TITLE
why?   a data preprocessing  problems in the use of keras.applications and keras-cv.

### ISSUE THREAD

**Issue by HLearning** [keras-team/keras-cv#892]
I found that some of the same weights were used in a and b

# keras.applications 
data preprocessing  
https://keras.io/api/applications/
```
img_path = 'elephant.jpg'
img = image.load_img(img_path, target_size=(224, 224))
x = image.img_to_array(img)
x = np.expand_dims(x, axis=0)
x = preprocess_input(x)
features = model.predict(x)
```
image imported to the network
- range: [0, 255]
- mode: BGR
- sub mean
- div std

# keras-cv models
https://keras.io/keras_cv/
```
densenet = keras_cv.models.DenseNet121(
  include_rescaling=True,
  include_top=True,
  classes=3
)
```
- include_rescaling=True,  image = image / 255

The two are different. In the process of use, does it have no impact？

**Comment by DavidLandup0** [comment]
It will have an impact - a major one. Here's what happens when you don't rescale and then rescale the same input on DenseNet from `keras.applications`:

```
url = 'https://upload.wikimedia.org/wikipedia/commons/0/02/Black_bear_large.jpg'
img = url_to_array(url)
img = preprocess_input(img)
img_batch = np.expand_dims(img, 0)

densenet = keras.applications.DenseNet121(weights='imagenet', include_top=True)
pred = densenet.predict(img_batch)
print(decode_predictions(pred))
# [[('n02133161', 'American_black_bear', 0.96903336), ('n02134418', 'sloth_bear', 0.025582273), ...
```

And:

```
url = 'https://upload.wikimedia.org/wikipedia/commons/0/02/Black_bear_large.jpg'
img = url_to_array(url)
img = preprocess_input(img)
img = img/255.0
img_batch = np.expand_dims(img, 0)

densenet = keras.applications.DenseNet121(weights='imagenet', include_top=True)
pred = densenet.predict(img_batch)
print(decode_predictions(pred))
# [[('n01930112', 'nematode', 0.030764641), ('n03291819', 'envelope', 0.022044677),
```

The weights were ported from `keras.applications` to `keras_cv.models` to avoid re-training an entire family of networks, which is expensive in time and resources since the network itself is the same. You can just set `include_rescaling` to `False` until the networks get re-trained (if they even do, since it'd kind of be a waste of computation), and it'll work just fine :)

**Comment by HLearning** [comment]
@DavidLandup0   Thank you for your answer. I understand what you said. 

I think scaling should be done in preprocessing, not in the backbone network. 

I think this interface(setup include_rescaling) should be dropout.

**Comment by HLearning** [comment]
I agree with you, but the confusion of apis has always been criticized by users. Now, in Keras, similar functions have different apis, which will make users very confused.

**Comment by LukeWood** [comment]
Sorry, this is a decision done for consistency across the board.  We don't really have the means to change it at this point due to API drift concerns.

If you consider this confusing, please pass `include_rescaling=False` in your model constructor.  Its a required argument on purpose - it makes users think about their inputs.  Inputting values in the range (0, 255) will not get good results unless you reduce your learning rate by 255 :).  It is important to gently nudge users towards the proper usage.
